flexible-event/index.js

```diff
@@ -0,0 +1,253 @@
+  let epsHigh = epsilonUpperBound
+
+  while (true) {
+    const epsilon = (epsHigh + epsLow) / 2
+    const infoGain = maxInformationGain(numStates, epsilon)
+
+    if (infoGain > infoGainUpperBound) {
+      epsHigh = epsilon
+      continue
+    }
+
+    // Allow slack by returning something slightly non-optimal (governed by the tolerance)
+    // that still meets the privacy bar. If epsHigh == epsLow we're now governed by the epsilon
+    // bound and can return.
+    if (infoGain < infoGainUpperBound - tolerance && epsHigh != epsLow) {
+      epsLow = epsilon
+      continue
+    }
+
+    return [epsilon, flipProbabilityDp(numStates, epsilon)]
+  }
+}
+
+function printConfigData(config, epsilon, sourceType) {
+  const numStates = numFlexibleStates(config)
+  const infoGain = maxInformationGain(numStates, epsilon)
+  const flipProb = flipProbabilityDp(numStates, epsilon)
+
+  console.log(`Number of possible different output states: ${numStates}`)
+  console.log(`Information gain: ${infoGain} bits`)
+  console.log(`Flip percent: ${100 * flipProb}%`)
+
+  const infoGainDefault = maxInformationGain(
+        numFlexibleStates(DEFAULT_CONFIG[sourceType]), epsilon)
+
+  if (infoGain > infoGainDefault) {
+      const [newEps, flipProb] = epsilonToBoundInfoGainAndDp(numStates, infoGainDefault, epsilon)
+      console.log(
+          `WARNING: info gain of ${infoGain} > ${infoGainDefault} for ${sourceType} sources. Would require a ${100 * flipProb}% flip chance (effective epsilon = ${newEps}) to resolve.`)
+  }
+}
+
+function parseSourceType(str) {
+  if (!str in DEFAULT_CONFIG) {
+    throw 'unknown source type'
+  }
+  return str
+}
+
+function commaSeparatedInts(str) {
+  return str.split(',').map(v => Number(v))
+}
+
+function getConfig(json, sourceType) {
+  const defaultMaxReports = sourceType == 'navigation' ? 3 : 1
+  const defaultWindows = sourceType == 'navigation' ? 3 : 1
+
+  const maxEventLevelReports =
+    ('max_event_level_reports' in json) ?  json.max_event_level_reports : defaultMaxReports
+
+  const topLevelNumWindows =
+    ('event_report_windows' in json) ?
+    json.event_report_windows.end_times.length : defaultWindows
+
+  const perTriggerDataConfigs = []
+  json.trigger_specs.forEach(spec => {
+    const numDataTypes = spec.trigger_data.length
+    const numWindows =
+      ('event_report_windows' in spec) ?
+      spec.event_report_windows.end_times.length : topLevelNumWindows
+
+      // Technically this can be larger, but we will always be constrained
+      // by `max_event_level_reports`.
+      const numBuckets =
+        ('summary_buckets' in spec) ?
+        spec.summary_buckets.length : maxEventLevelReports
+
+      for (let i = 0; i < numDataTypes; i++) {
+        perTriggerDataConfigs.push([numWindows, numBuckets])
+      }
+  });
+
+  return {maxEventLevelReports, perTriggerDataConfigs}
+}
+
+const optionDefs = [
+  {
+    name: 'max_event_level_reports',
+    alias: 'm',
+    type: Number,
+    defaultValue: 20,
```

Comment:
The default should be based on the source type

Reply:
The original code used 20, no?

Reply:
Ah yeah, sorry this is fine.